### PR TITLE
[freeboxos] Fix phone and calls configuration

### LINKED ISSUE #17262: [freeboxos] Phone / calls: mismatch with default refresh interval
Default is set to 30s for phone while it is set to 2s for calls.
It is the reverse compared to the original binding.
The idea to have the short refresh interval was to detect quickly a phone suddenly ringing. This is a channel on the phone thing. So the short refresh should be on fvs thing type.
There is no hurry to refresh last call.
By the way, the short refresh is interesting only if the `ringing` channel is linked. So maybe we could have a default refresh set to 30s and the user will reduce it if he links the `ringing` channel. WDYT @clinique ?

PS: what does "FXS" mean for ? It is too late to change thing type but at least the label of discovered thing should have a label that talks to any user.

### PR DESCRIPTION
**Pull request by lolodomo**
Fix #17262

Also remove the unused class LanlineConfiguration

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Review by clinique — APPROVED** [review]

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/config/PhoneConfigurationBuilder.java [hunk 1]
```diff
@@ -42,13 +42,20 @@ public static PhoneConfigurationBuilder getInstance() {
     }
 
     public DiscoveryResultBuilder configure(ThingUID bridgeUID, Status config) {
-        ThingUID thingUID = new ThingUID(Type.DECT.equals(config.type()) ? THING_TYPE_DECT : THING_TYPE_FXS, bridgeUID,
-                Integer.toString(config.id()));
+        ThingUID thingUID;
+        String label;
+        if (Type.DECT.equals(config.type())) {
+            thingUID = new ThingUID(THING_TYPE_DECT, bridgeUID, Integer.toString(config.id()));
+            label = "DECT Phone";
```
Comment (jlaur) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/config/PhoneConfigurationBuilder.java:49]: Perhaps we should support I18N for this label?

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/config/PhoneConfigurationBuilder.java [hunk 2]
```diff
@@ -42,13 +42,20 @@ public static PhoneConfigurationBuilder getInstance() {
...
+        } else {
+            thingUID = new ThingUID(THING_TYPE_FXS, bridgeUID, Integer.toString(config.id()));
+            label = "Landline Phone";
```
Comment (jlaur) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/config/PhoneConfigurationBuilder.java:52]: Same here.

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/discovery/FreeboxOsDiscoveryService.java [hunk 3]
```diff
@@ -159,10 +159,10 @@ private void discoverPhone(ThingUID bridgeUID) {
             logger.debug("Error discovering phones: {}", e.getMessage());
         }
         if (!statuses.isEmpty()) {
-            ThingUID thingUID = new ThingUID(THING_TYPE_CALL, bridgeUID, "landline");
+            ThingUID thingUID = new ThingUID(THING_TYPE_CALL, bridgeUID, "calls");
             logger.debug("Adding new Call thing {} to inbox", thingUID);
             DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withBridge(bridgeUID)
-                    .withLabel("Freebox Calls").build();
+                    .withLabel("Phone Calls").build();
```
Comment (jlaur) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/discovery/FreeboxOsDiscoveryService.java:165]: I18N support?

**Review by jlaur — COMMENTED** [review]
Thanks, a few I18N questions, otherwise LGTM.

**Comment by lolodomo** [comment]
@jlaur : I would like to introduce i18n for label of discovered things in a separate PR if you are ok.

**Comment by jlaur** [comment]
> I would like to introduce i18n for label of discovered things in a separate PR if you are ok.

Sure!

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**